.buildkite/scripts/pull-request-pipeline/generate-steps.py

```diff
@@ -0,0 +1,105 @@
+import os
+import requests
+import sys
+import typing
+from requests.adapters import HTTPAdapter, Retry
+
+from ruamel.yaml import YAML
+
+RELEASES_URL = "https://raw.githubusercontent.com/elastic/logstash/main/ci/logstash_releases.json"
+TEST_MATRIX_URL = "https://gist.githubusercontent.com/mashhurs/c5b66e05eac1c0968f841e489a8b43a6/raw/eb6b19174977aa4fbe6f7309e35481fb31690e88/tes.yaml"
```

Comment:
review note: this will be updated once https://github.com/elastic/logstash-filter-elastic_integration/pull/202 is landed. This is a test that actual CI steps will be seen here.